mcs/class/corlib/System/WeakAttribute.cs

```diff
@@ -0,0 +1,12 @@
+using System;
+
+#if MOBILE
+namespace System {
+
+[AttributeUsage(AttributeTargets.Field)]
+public sealed class WeakAttribute : Attribute
```

Comment:
I still think the name is very common, this can be easily breaking change due to ambiguity. We should rename it to `WeakReferenceAttribute`

Reply:
It might also be useful to add sample into this file how it's intended to be used